include/envoy/ssl/context_config.h

```diff
@@ -60,6 +60,7 @@ class ContextConfig {
    */
   virtual const std::string& certificateRevocationListPath() const PURE;
 
+  // TODO(lizan): consider refactor 4 methods below to Ssl::TlsCertificateConfig
```

Comment:
+1 for this change.

Reply:
FWIW: https://github.com/lizan/envoy/compare/static_secret_provider...tls_certificate_config , I prefer to have this change in a follow up PRs, but if reviewers prefer in same PR, I will add a commit.